packages/sitecore-jss-react/src/components/PlaceholderCommon.tsx

```diff
@@ -131,34 +147,41 @@ export class PlaceholderCommon<T extends PlaceholderProps> extends React.Compone
 
   static getPlaceholderDataFromRenderingData(
     rendering: ComponentRendering | RouteData,
-    name: string
+    name: string,
+    editMode?: EditMode
   ) {
     let result;
+    let phName = name.slice();
+
     /** [SXA] it needs for deleting dynamics placeholder when we set him number(props.name) of container.
     from backend side we get common name of placeholder is called 'nameOfContainer-{*}' where '{*}' marker for replacing **/
     if (rendering?.placeholders) {
       Object.keys(rendering.placeholders).forEach((placeholder) => {
-        const patternPlaceholder =
-          placeholder.indexOf('{*}') !== -1
-            ? new RegExp(`^${placeholder.replace(/\{\*\}+/i, '\\d+')}$`)
-            : null;
-
-        if (patternPlaceholder && patternPlaceholder.test(name)) {
-          rendering.placeholders[name] = rendering.placeholders[placeholder];
-          delete rendering.placeholders[placeholder];
+        const patternPlaceholder = isDynamicPlaceholder(placeholder)
+          ? getDynamicPlaceholderPattern(placeholder)
+          : null;
+
+        if (patternPlaceholder && patternPlaceholder.test(phName)) {
+          // When working in Edit Mode Metadata we need to keep the raw placeholder name
```

Comment:
Can you please expand the comments (either here, in PlaceholderMetadata or unit tests) to describe the usecase for dynamic placeholders, and the differences on how they are handled in chrome vs metadata modes?

Reply:
Updated